### ISSUE TITLE
Contain cluster/node surge upgrade max surge/max unavailable defaults to unusable values

### ISSUE THREAD

**Issue by brettcurtis** [hashicorp/terraform-provider-google#17164]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team)
  to expedite investigation and resolution of this issue.

### Terraform Version

Terraform v1.7.2

### Affected Resource(s)

google_container_cluster (autoscaler)
google_container_node_pool

### Expected Behavior

max_surge and max_unavailable default to 0/0 and when using surge:

![image](https://github.com/hashicorp/terraform-provider-google/assets/1610100/10a28116-49ea-42d2-a6f4-0ad90428d47b)

Should the default work, maybe this isn't a norm but it's the first time I think I've seen it so I figured I'd toss it out here. Feel free to close it out if this is common behavior.

b/323650143

**Comment by edwardmedia** [comment]
Based on the [doc](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/UpgradeSettings), defaults to 0 for both attributes are questionable

**Comment by jasonkramberger** [comment]
@brettcurtis is this in manually created or NAP created node pools?

I do see that max_surge is Computed in the node_pool resource but just Optional in the cluster autoscaler schema.

**Comment by brettcurtis** [comment]
You're correct this is just for NAP created pools, standard node pools look good.
